### PR TITLE
Feature: Update datamodels 

### LINKED ISSUE #15: Add dataclasses for models

### PR DESCRIPTION
**Pull request by anisbhsl**
### Major Changes:

Added `SinequaBaseModel` as abstract class. All other model classes now inherit this abstract class and have to implement `generate_payload(..)` method.

### Minor Changes
1. Added dataclasses
2. Added a test to check queryparams payload generation
3. Code fixes

### CODE REVIEW

File: pynequa/models.py [hunk 1]
```diff
@@ -1,13 +1,48 @@
 from typing import Dict, List, Optional
-
-
-class TreeParams:
+from abc import abstractmethod
+from dataclasses import dataclass, field
+import logging
+
+logging.basicConfig(level=logging.DEBUG)
```
Comment (NISH1001) [inline pynequa/models.py:6]: Let's use [loguru](https://github.com/Delgan/loguru) for logging. In-built logging module is pretty obfuscated and difficult to navigate.

File: pynequa/models.py [hunk 2]
```diff
@@ -1,13 +1,48 @@
...
+    def generate_payload(self, **kwargs) -> Dict:
+        """
+        This is abstract method for SinequaBaseModel.
+        Every child class should implement this method.
+        """
+        raise NotImplementedError()
+
+
+@dataclass
+class TreeParams(SinequaBaseModel):
```
Comment (NISH1001) [inline pynequa/models.py:24]: Isn't the base model and Params/dataclass different? What's the difference?

File: pynequa/models.py (hunk 1 above)
Comment (anisbhsl) [reply pynequa/models.py:6]: I wanted to use in-built module as much as possible just to keep this package at its bare minimum. But I can change it to use loguru.

File: pynequa/models.py [hunk 3]
```diff
@@ -1,13 +1,48 @@
...
-
-class TreeParams:
+from abc import abstractmethod
+from dataclasses import dataclass, field
+import logging
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+class SinequaBaseModel:
```
[the remark this answers is not in the document]
Comment (anisbhsl) [reply pynequa/models.py:9]: Sure. Thanks

File: pynequa/models.py (hunk 2 above)
Comment (anisbhsl) [reply pynequa/models.py:24]: I think the naming might have confused you which I just realized as well. `SinequaBaseModel` is more of a base class that will have methods that other `Params` class will implement for now as there aren't any shared attributes which could be used across all.

**Comment by NISH1001** [comment]
LGTM

**Pull request merged**